nixos/lib/make-disk-image.nix

```diff
@@ -70,6 +82,22 @@
 , # Disk image format, one of qcow2, qcow2-compressed, vdi, vpc, raw.
   format ? "raw"
 
+  # Whether to fix:
+  #   - GPT Disk Unique Identifier (diskGUID)
+  #   - GPT Partition Unique Identifier: depends on the layout, root partition UUID can be controlled through `rootGPUID` option
+  #   - GPT Partition Type Identifier: fixed according to the layout, e.g. ESP partition, etc. through `parted` invocation.
+  #   - Filesystem Unique Identifier when fsType = ext4 for *root partition*.
+  # BIOS/MBR support is "best effort" at the moment.
+  # Boot partitions may not be deterministic.
+  # Also, to fix last time checked of the ext4 partition if fsType = ext4.
+, deterministic ? true
```

Comment:
Is there a case where we don't want deterministic file images?

Reply:
I assume we might want disk images with different predictable UUIDs or we want to avoid a specific UUIDs.
In all cases, I do not offer the possibility to configure all IDs, therefore, I feel like we should offer this option.

Reply:
Also, I assume deterministic mode could introduce unwanted side effects, turning it off would be an escape hatch for users, but I do not have any example on top of my mind.